### PR TITLE
Changes needed to implement legacy PN9 operators

### PR DESCRIPTION
**Pull request by natsegal**
add annotation files
add metadata to the batch

### CODE REVIEW

File: nmtwizard/preprocess/loader.py [hunk 1]
```diff
@@ -100,22 +100,31 @@ def close_files(self):
...
+            if not tu_list:
                 return
 
-            yield tu_batch
+            meta = self._file.weight.copy()
+            meta["base_name"] = self._file.base_name
+            meta["root"] = self._file.root
+            meta["no_preprocess"] = self._file.no_preprocess
+
+            yield tu_list, meta
```
Comment (guillaumekln) [inline nmtwizard/preprocess/loader.py]: What is the difference between this meta and the one included in the `TranslationUnit` class?

File: nmtwizard/preprocess/sampler.py [hunk 2]
```diff
@@ -72,8 +101,23 @@ def _discover_files():
...
+                source_dir = d_item["path"]
 
             distribution = d_item['distribution']
+
+            # Get annotations
+            annotations = d_item.get('annotations', {})
+            if not isinstance(annotations, dict):
+                raise ValueError("invalid type for 'annotations' field")
+            for key in annotations:
+                if not isinstance(annotations[key], str):
```
Comment (guillaumekln) [inline nmtwizard/preprocess/sampler.py]: The key lookup should be computed once, before the conditions.

File: nmtwizard/preprocess/loader.py (hunk 1 above)
Comment (natsegal) [reply nmtwizard/preprocess/loader.py]: Meta here is batch-level, for the moment it mostly gives the information in training about the file the batch comes from.
Meta in translation unit is about the particular example. It contains context information for example.
Maybe I could find more sensible names for both...

**Comment by natsegal** [comment]
I tested annotations from systran docker project, but you are right, those tests belong here, I'll move them.

**Pull request merged**